File: modified_scripts/c/binary_search/deepseek-coder_6.7b.c
int binary_search(int arr[], int start, int end, int x) {
    if (end >= start) {
        int mid = start + (end - start) / 2;
        if (arr[mid] == x) return mid;
        if (arr[mid] > x) return binary_search(arr, start, mid-1, x);
        return binary_search(arr, mid+1, end, x);
    }
    return -1;
}

File: modified_scripts/r/binary_search/deepseek-coder_6.7b.r

binary_search <- function(arr, target) {
    left = 1
    right = length(arr)
    
    while (left <= right) {
        mid = ceiling((right + left)/2)
        
        if (target == arr[mid]) {
            return(mid) 
        } else if (target < arr[mid]) {
            right = mid - 1
        } else {
            left = mid + 1
        }
    }
    
    return(-1)  
}